frontend/catalyst/cuda/__init__.py

```diff
@@ -16,12 +16,20 @@
 entry point.
 """
 
+from importlib.metadata import version
 from pathlib import Path
 
 import cudaq
 import pennylane as qml
 
-from catalyst.cuda.catalyst_to_cuda_interpreter import interpret
+
+def _check_version_compatibility():
+    installed_version = version("cuda_quantum")
+    compatible_version = "0.6.0"
+    if installed_version != compatible_version:
+        msg = f"Compiling with incompatible version cuda_quantum=={installed_version}."
+        msg += f"Please install compatible version cuda_quantum=={compatible_version}."
+        raise ValueError(msg)
```

Comment:
```suggestion
        raise ModuleNotFoundError(msg)
```
Would it be better to raise ModuleNotFoundError? (I am not sure)

Reply:
Yeah, I was not sure either :sweat_smile: I'm happy to make the change. Thanks Sergei!